app/models/spree/order_decorator.rb

```diff
@@ -0,0 +1,14 @@
+module SpreeGateway
+  module OrderDecorator
+    def confirmation_required?
+      return false if paid_with_apple_pay?
+      super
+    end
+
+    def paid_with_apple_pay?
+      payments.any? { |payment| payment.payment_method.is_a? Spree::Gateway::StripeApplePayGateway }
```

Comment:
Let's rewrite it as `payments.any?(&:apple_pay?)` and move `is_a?` to a `PaymentDecorator` so that each component has a single responsibility

Reply:
fixing